### PR TITLE
Mod0

### PR DESCRIPTION
**Pull request by kcny**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+  helper_method :current_user
+
+  def require_user
+    redirect_to sign_in_path unless current_user
+  end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```
Comment (brandoncc) [inline app/controllers/application_controller.rb]: You should memoize this. This is something that will probably be called multiple times on each request. You don't want each call to hit the db.

File: app/views/sessions/new.html.haml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+%section.sign_in.container
+  .row
+    .col-sm-10.col-sm-offset-1
+      %form.sign_in
+      = bootstrap_form_tag url: '/login', html: {class: 'sign_in'} do |f|
```
Comment (brandoncc) [inline app/views/sessions/new.html.haml]: You should use named route helpers, like `login_path`. That way if you ever decide to change the url, you don't have to manually search all of your code. You will only have to change your routes file.

File: app/views/sessions/new.html.haml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        %header
+          %h1 Sign In
+        %fieldset
+          = label_tag     :email, "Email Address"
```
Comment (brandoncc) [inline app/views/sessions/new.html.haml]: You should be using the form builder that is provided by `bootstrap_form_tag`. You use it like this `f.email_field`. That will provide you a label as well, which you can read the docs for information on how to customize the label text.

File: app/views/users/new.html.haml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+%section.register.container
+  .row
+    .col-sm-10.col-sm-offset-1
+      = bootstrap_form_for @user, html: { class: "form-horizontal"} do |f|
+        %header
+          %h1 Register
+        %fieldset
+          = f.email_field   :email, label: "Email Address"
```
Comment (brandoncc) [inline app/views/users/new.html.haml]: This is how you should have the one I referenced above written.

File: config/routes.rb [hunk 5]
```diff
@@ -1,8 +1,21 @@
 Myflix::Application.routes.draw do
-  get 'ui(/:action)', controller: 'ui'
-  get '/home', to: 'videos#index'
+  root to: 'pages#front'
+  get 'register',       to: 'users#new'
+  get '/sign_in',       to: 'sessions#new'
```
Comment (brandoncc) [inline config/routes.rb]: fyi, you get the named route helpers by adding `as` to the custom routes.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (kcny) [inline app/controllers/application_controller.rb]: Will research and follow with question.

File: app/views/sessions/new.html.haml (hunk 2 above)
Comment (kcny) [inline app/views/sessions/new.html.haml]: Having trouble getting this to work.  Will follow up for clarification.

File: app/views/sessions/new.html.haml (hunk 3 above)
Comment (kcny) [inline app/views/sessions/new.html.haml]: Okay, got it.  Thanks!

File: config/routes.rb (hunk 5 above)
Comment (kcny) [inline config/routes.rb]: I've updated the get  'sign_in' to get 'login' for consistency.  Had a few questions on this and will follow up during your office hours. Thanks for reviewing. Cheers!

File: config/routes.rb [hunk 6]
```diff
@@ -1,8 +1,21 @@
...
+  resources :videos, except: [:destroy] do 
+    collection do
+      post :search, to: "videos#search"
+    end
+  end
+  get 'ui(/:action)', controller: 'ui'
   resources :categories, only: [:show]
-end
+  resources :users, only: [:create] 
+  resources :sessions, ony: [:create]
```
Comment (ppj) [inline config/routes.rb]: Do you need this line when you already have defined the individual route (line 5)?

**Pull request merged**